src/CommunityToolkit.Maui.Camera/CameraManager.android.cs

```diff
@@ -0,0 +1,353 @@
+
+	NativePlatformCameraPreviewView? previewView;
+	IExecutorService? cameraExecutor;
+	ProcessCameraProvider? processCameraProvider;
+	ImageCapture? imageCapture;
+	ImageCallBack? imageCallback;
+	ICamera? camera;
+	ICameraControl? cameraControl;
+	Preview? cameraPreview;
+	ResolutionSelector? resolutionSelector;
+	ResolutionFilter? resolutionFilter;
+
+	public void Dispose()
+	{
+		Dispose(true);
+		GC.SuppressFinalize(this);
+	}
+
+	// IN the future change the return type to be an alias
+	public NativePlatformCameraPreviewView CreatePlatformView()
+	{
+		imageCallback = new ImageCallBack(cameraView);
+		previewView = new NativePlatformCameraPreviewView(context);
+		if (NativePlatformCameraPreviewView.ScaleType.FitCenter is not null)
+		{
+			previewView.SetScaleType(NativePlatformCameraPreviewView.ScaleType.FitCenter);
+		}
+		cameraExecutor = Executors.NewSingleThreadExecutor() ?? throw new CameraViewException($"Unable to retrieve {nameof(IExecutorService)}");
+
+		return previewView;
+	}
+
+	public partial void UpdateFlashMode(CameraFlashMode flashMode)
+	{
+		if (imageCapture is null)
+		{
+			return;
+		}
+
+		imageCapture.FlashMode = flashMode.ToPlatform();
+	}
+
+	public partial void UpdateZoom(float zoomLevel)
+	{
+		if (cameraControl is null)
+		{
+			return;
+		}
+
+		cameraControl.SetZoomRatio(zoomLevel);
+	}
+
+	public async partial ValueTask UpdateCaptureResolution(Size resolution, CancellationToken token)
+	{
+		if (resolutionFilter?.TargetSize.Width == resolution.Width && resolutionFilter?.TargetSize.Height == resolution.Height)
+		{
+			return;
+		}
+
+		var targetSize = new Android.Util.Size((int)resolution.Width, (int)resolution.Height);
+
+		if (resolutionFilter is null)
+		{
+			resolutionFilter = new ResolutionFilter(targetSize);
+		}
+		else
+		{
+			resolutionFilter.TargetSize = targetSize;
+		}
+
+		resolutionSelector?.Dispose();
+
+		resolutionSelector = new ResolutionSelector.Builder()
+		.SetAllowedResolutionMode(ResolutionSelector.PreferHigherResolutionOverCaptureRate)
+		.SetResolutionFilter(resolutionFilter)
+		.Build();
+
+		if (IsInitialized)
+		{
+			await StartUseCase(token);
+		}
+	}
+
+	protected virtual void Dispose(bool disposing)
+	{
+		if (disposing)
+		{
+			camera?.Dispose();
+			camera = null;
+
+			cameraControl?.Dispose();
+			cameraControl = null;
+
+			cameraPreview?.Dispose();
+			cameraPreview = null;
+
+			cameraExecutor?.Dispose();
+			cameraExecutor = null;
+
+			imageCapture?.Dispose();
+			imageCapture = null;
+
+			imageCallback?.Dispose();
+			imageCallback = null;
+
+			previewView?.Dispose();
+			previewView = null;
+
+			processCameraProvider?.Dispose();
+			processCameraProvider = null;
+
+			resolutionSelector?.Dispose();
+			resolutionSelector = null;
+
+			resolutionFilter?.Dispose();
+			resolutionFilter = null;
+		}
+	}
+
+	protected virtual async partial Task PlatformConnectCamera(CancellationToken token)
```

Comment:
Let's punt this to a future PR because we have many methods that return `ValueTask` and `Task` that also accept a `CancellationToken` without a default parameter. 

I agree with you. But let's keep the codebase consistent for this PR, and in a future PR we can add the default parameter everywhere.